pkg/config/msp_test.go

```diff
@@ -947,24 +947,53 @@ func TestCreateApplicationMSPCRL(t *testing.T) {
 		ChannelGroup: channelGroup,
 	}
 
-	c := New(config)
+	originalConfigTx := New(config)
 
-	org1MSP, err := c.ApplicationMSP("Org1")
+	// update org2MSP to include an intemediate cert that is different
+	// from the root cert
+	org2MSP, err := originalConfigTx.ApplicationMSP("Org2")
 	gt.Expect(err).NotTo(HaveOccurred())
-
-	cert, privKey, _ := certPrivKeyCRL(org1MSP)
-	certToRevoke, _ := generateCertAndPrivateKeyFromCACert(t, "org1.example.com", cert, privKey)
-	signingIdentity := &SigningIdentity{
-		Certificate: cert,
-		PrivateKey:  privKey,
-		MSPID:       "MSPID",
-	}
-	newCRL, err := c.CreateApplicationMSPCRL("Org1", signingIdentity, certToRevoke)
+	org2Cert, org2PrivKey, _ := certPrivKeyCRL(org2MSP)
+	org2IntermediateCert, org2IntermediatePrivKey := generateIntermediateCACertAndPrivateKey(t, "org2.example.com", org2Cert, org2PrivKey)
+	org2MSP.IntermediateCerts = append(org2MSP.IntermediateCerts, org2IntermediateCert)
+	err = originalConfigTx.UpdateApplicationMSP(org2MSP, "Org2")
 	gt.Expect(err).NotTo(HaveOccurred())
-	gt.Expect(newCRL.TBSCertList.RevokedCertificates).To(HaveLen(1))
-	gt.Expect(newCRL.TBSCertList.RevokedCertificates[0].SerialNumber).To(Equal(certToRevoke.SerialNumber))
 
-	// TODO create a CRL using an intermediate cert
+	t.Run("create CRL using a root cert", func(t *testing.T) {
+		t.Parallel()
+		gt := NewGomegaWithT(t)
+		c := New(originalConfigTx.Updated())
+		org1MSP, err := c.ApplicationMSP("Org1")
+		gt.Expect(err).NotTo(HaveOccurred())
+
+		cert, privKey, _ := certPrivKeyCRL(org1MSP)
+		certToRevoke, _ := generateCertAndPrivateKeyFromCACert(t, "org1.example.com", cert, privKey)
+		signingIdentity := &SigningIdentity{
+			Certificate: cert,
+			PrivateKey:  privKey,
+			MSPID:       "MSPID",
+		}
+		org1CRL, err := c.CreateApplicationMSPCRL("Org1", signingIdentity, certToRevoke)
+		gt.Expect(err).NotTo(HaveOccurred())
+		gt.Expect(org1CRL.TBSCertList.RevokedCertificates).To(HaveLen(1))
+		gt.Expect(org1CRL.TBSCertList.RevokedCertificates[0].SerialNumber).To(Equal(certToRevoke.SerialNumber))
+	})
+
+	t.Run("create CRL using an intermediate cert", func(t *testing.T) {
+		t.Parallel()
+		gt := NewGomegaWithT(t)
+		c := New(originalConfigTx.Updated())
+		signingIdentity := &SigningIdentity{
+			Certificate: org2IntermediateCert,
+			PrivateKey:  org2IntermediatePrivKey,
+			MSPID:       "MSPID",
+		}
+		certToRevoke, _ := generateCertAndPrivateKeyFromCACert(t, "org2.example.com", org2IntermediateCert, org2IntermediatePrivKey)
+		org2CRL, err := c.CreateApplicationMSPCRL("Org2", signingIdentity, certToRevoke)
+		gt.Expect(err).NotTo(HaveOccurred())
+		gt.Expect(org2CRL.TBSCertList.RevokedCertificates).To(HaveLen(1))
+		gt.Expect(org2CRL.TBSCertList.RevokedCertificates[0].SerialNumber).To(Equal(certToRevoke.SerialNumber))
```

Comment:
I think that this is not enough and you should also call [CheckCRLSignature](https://golang.org/pkg/crypto/x509/#Certificate.CheckCRLSignature) to ensure the CRL was properly signed by the intermediate CA. 

The same thing should be done for root CAs without intermediate CAs, that issue leaf certificates.

Reply:
Good call. Definitely makes the test more thorough. Done!